File: LaunchModeDemo/app/src/main/java/com/example/sakura/launchmodedemo/SecondActivity.java
package com.example.sakura.launchmodedemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

/**
 * Created by sakura on 2018/6/1.
 */

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        findViewById(R.id.btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(SecondActivity.this, MainActivity.class);
                intent.putExtra("data", "Data from second");
                startActivity(intent);
            }
        });
    }
}

File: DisposableDemo/app/src/main/java/com/example/sakura/disposabledemo/SecondActivity.kt
package com.example.sakura.disposabledemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import java.util.concurrent.TimeUnit

class SecondActivity: AppCompatActivity() {

    private var disposable = CompositeDisposable()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_second)

        disposable.add(Observable.timer(2, TimeUnit.SECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { Log.d("test", "$it") })
    }

    override fun onStart() {
        super.onStart()

        disposable.add(Observable.timer(3, TimeUnit.SECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { Log.d("test", "$it") })
    }

    override fun onResume() {
        super.onResume()

        disposable.add(Observable.timer(4, TimeUnit.SECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { Log.d("test", "$it") })
    }

    override fun onDestroy() {
        disposable.clear()

        super.onDestroy()
    }

}